aten/src/ATen/native/cuda/layer_norm_kernel.cu

```diff
@@ -1116,6 +1121,18 @@ void LayerNormKernelImplInternal(
   }
 }
 
+inline bool use_cudnn_layernorm(bool gamma, bool beta, at::ScalarType dtype) {
+#ifndef USE_ROCM
+  static bool enabled = (c10::utils::check_env("TORCH_CUDNN_LAYERNORM_ENABLED") == true);
+  // cuDNN only supports bf16, fp16, and fp32
+  bool dtype_supported = dtype != at::ScalarType::Double;
```

Comment:
This feels very unsafe. If cudnn indeed only supports those 3 dtypes, please check for them as this funciton will return true if someone passes say one of fp8 dtypes